packages/react-centra-checkout/src/server/createGetCentraWebhookEvents.ts

```diff
@@ -0,0 +1,166 @@
+import type { NextApiRequest } from 'next'
+import crypto from 'node:crypto'
+import { z } from 'zod'
+
+const sign = (secret: string, dataString: string) => {
+  const hmac = crypto.createHmac('sha256', secret)
+
+  hmac.update(dataString)
+
+  return hmac.digest('hex')
+}
+
+/**
+ * Parse a named parameters string into an object.
+ * @example
+ * parseNamedParametersString('foo=bar,baz=qux') // \{ foo: 'bar', baz: 'qux' \}
+ */
+const parseNamedParametersString = (input: string): Record<string, string> => {
+  return input.split(',').reduce((acc, pair) => {
+    const [key, value] = pair.split('=')
+    if (!key) {
+      return acc
+    }
+
+    return { ...acc, [key]: value }
+  }, {})
+}
+
+const bodyParserSchema = z
+  .object({ payload: z.string() })
+  .nullable()
+  .catch(() => null)
+
+type EventTypes =
+  | 'affiliates'
+  | 'brands'
+  | 'campaignSites'
+  | 'campaigns'
+  | 'categories'
+  | 'cmsArticles'
+  | 'giftCertificates'
+  | 'mapsLocations'
+  | 'mapsRegion'
+  | 'markets'
+  | 'pricelists'
+  | 'products'
+  | 'statics'
+  | 'warehouseGroups'
+  | 'warehouses'
+
+type CreateGetCentraWebhookEventsConfig<TRequest> = {
+  adapter: {
+    isRequestMethodPost?: (request: TRequest) => boolean
+    getHeader: (request: TRequest, headerKey: string) => string | null | undefined
+    getRawBody: (request: TRequest) => unknown
+  }
+  secret?: string
+}
+
+/**
+ * Create get Centra webhook events.
+ * Parses and returns the events passed from Centra.
+ *
+ * @see https://centra.dev/docs/services/centra-webhooks
+ * @example
+ * ```typescript
+ * const getCentraWebhookEvents = createGetCentraWebhookEvents({
+ *   adapter: nextAppRouterAdapter
+ * })
+ *
+ * const POST = async (req: Request) => {
+ *   const [error, results] = await getCentraWebhookEvents(req)
+ *
+ *   if (error) {
+ *     // Handle error codes
+ *     return Response.json({ message: 'Error' }, { status: 500 })
+ *   }
+ *
+ *   if ('products' in results) {
+ *     return Response.json({ products: results.products }, { status: 200 })
+ *   }
+ * }
+ * ```
+ */
+export function createGetCentraWebhookEvents<TRequest>(
```

Comment:
Consider an explicit return type here to avoid an excess of `as const` assertions. Changes to this function could inadvertently change the return type without somebody noticing.

Reply:
E.g `export function createGetCentraWebhookEvents<TRequest>(config: CreateGetCentraWebhookEventsConfig<TRequest> ): ResponseType { ... }`